plugins/janus_videoroom.c

```diff
@@ -7489,14 +7503,41 @@ static void janus_videoroom_recorder_create(janus_videoroom_publisher *participa
 		participant->drc = rc;
 	}
 }
+// CARBYNE-S3-UPLOAD
+
+int create_recording_json(char *filename, char *codec)
+{
+	char *jsonObj;
+	asprintf(&jsonObj, "{\n \"recordingName\" : \"%s\" ,\n \"directory\" : \"/home/ubuntu/recordings/\" ,\n\"codec\" : \"%s\"\n}", filename, codec);
+	char *fullFileName;
+	// removeSubstr(filename,".mjr");
+	asprintf(&fullFileName, "/home/ubuntu/RecJSON/%s.json", filename);
+	printf("[file debug]------%s\n", fullFileName);
+	FILE *fp;
+
+	fp = fopen(fullFileName, "w+");
 
+	fputs(jsonObj, fp);
+
+	fclose(fp);
+
+	return 0;
+}
+// CARBYNE-S3-UPLOAD-END
 static void janus_videoroom_recorder_close(janus_videoroom_publisher *participant)
 {
 	if (participant->arc)
 	{
 		janus_recorder *rc = participant->arc;
 		participant->arc = NULL;
 		janus_recorder_close(rc);
+		// upload file
+		// upload_recording(rc->filename,rc->codec);
+		printf("[file debug]------calling create_recording_json\n");
```

Comment:
Does printf go to wgw logs? cloud watch?

Reply:
yes